src/test/java/io/confluent/kafkarest/mock/MockKafkaConsumer.java

```diff
@@ -0,0 +1,236 @@
+package io.confluent.kafkarest.mock;
+
+
+import io.confluent.kafkarest.Time;
+import io.confluent.kafkarest.entities.ConsumerRecord;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.consumer.OffsetCommitCallback;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Assert;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+
+public class MockKafkaConsumer implements Consumer<byte[], byte[]> {
```

Comment:
There's also a `MockConsumer` included with Kafka (that ships in the main clients jar). Would that be sufficient  so we could avoid maintaining a separate mock implementation here?